corehq/messaging/smsbackends/vertex/models.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+import requests
+import re
+
+from corehq.messaging.smsbackends.http.models import SQLSMSBackend
+from corehq.messaging.smsbackends.vertex.forms import VertexBackendForm
+from corehq.apps.sms.util import strip_plus
+
+VERTEX_URL = "https://www.smsjust.com/sms/user/urlsms.php"
+SUCCESS_RESPONSE_REGEX = r'^(\d+)-(20\d{2})_(\d{2})_(\d{2})$'  # 570737298-2017_05_27
+SUCCESS_RESPONSE_REGEX_MATCHER = re.compile(SUCCESS_RESPONSE_REGEX)
+
+
+class VertexBackend(SQLSMSBackend):
+    class Meta:
+        app_label = 'sms'
+        proxy = True
+
+    @classmethod
+    def get_api_id(cls):
+        return 'VERTEX'
+
+    @classmethod
+    def get_generic_name(cls):
+        return "Vertex"
+
+    @classmethod
+    def get_available_extra_fields(cls):
+        return [
+            'username',
+            'password',
+            'senderid',
+        ]
+
+    @classmethod
+    def get_form_class(cls):
+        return VertexBackendForm
+
+    def populate_params(self, msg_obj):
+        config = self.config
+        params = {
+            'username': config.username,
+            'pass': config.password,
+            'senderid': config.senderid,
+            'response': 'Y',
+            # It must include the country code appended before the mobile number
+            # The mobile number should contain only numbers and no symbols like "+", "-" etc.
+            'dest_mobileno': strip_plus(msg_obj.phone_number),
+            'msgtype': 'UNI'  # TXT/UNI/FLASH/WAP
```

Comment:
we shouldn't default to unicode - when you send a message as unicode, the message is charged in 70-character chunks (since each character takes up 16 bits). we should instead do something like:

    try:
        params['message'] = str(msg_obj.text)
        params['msgtype'] = 'TXT'
    except UnicodeEncodeError:
        params['message'] = msg_obj.text.encode('utf-8')
        params['msgtype'] = 'UNI'